assignment-1/pkg/apis/shahpratikr.dev/v1alpha1/types.go

```diff
@@ -0,0 +1,32 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Snapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              SnapshotSpec   `json:"spec,omitempty"`
+	Status            SnapshotStatus `json:"status,omitempty"`
+}
+
+type SnapshotSpec struct {
+	PvcName   string `json:"pvcname,omitempty"`
+	Operation string `json:"operation,omitempty"`
+}
+
+type SnapshotStatus struct {
+	VolumeSnapshotName string `json:"volumesnapshotname,omitempty"`
+	Storage            string `json:"storage,omitempty"`
+	StorageClassname   string `json:"storageclassname,omitempty"`
```

Comment:
We should add a field to set the status of the operation. Whether it is failed/succeeded/in-progress

Reply:
I need to read about how to update status. I tried to add a function which updates status. but was not able to do it properly.